custom_model_runner/datarobot_drum/drum/typeschema_validation.py

```diff
@@ -278,18 +298,27 @@ def validate(self, dataframe):
             logger.info("Skipping type validation")
             return []
         types = dict()
-        types[Values.NUM] = dataframe.select_dtypes(np.number).shape[1] > 0
-        txt_columns = self.number_of_text_columns(dataframe)
-        img_columns = self.number_of_img_columns(dataframe)
-        types[Values.TXT] = txt_columns > 0
-        types[Values.IMG] = img_columns > 0
-        types[Values.CAT] = (
-            dataframe.select_dtypes("O").shape[1]
-            - (txt_columns + img_columns)
-            + dataframe.select_dtypes("boolean").shape[1]
-            > 0
-        )
-        types[Values.DATE] = dataframe.select_dtypes("datetime").shape[1] > 0
+
+        if check_is_sparse(dataframe):
+            # only numeric can be a csr or matrix market sparse matrix
+            types[Values.NUM] = True
+            types[Values.TXT] = False
+            types[Values.IMG] = False
+            types[Values.CAT] = False
+            types[Values.DATE] = False
+        else:
+            types[Values.NUM] = dataframe.select_dtypes(np.number).shape[1] > 0
+            txt_columns = self.number_of_text_columns(dataframe)
+            img_columns = self.number_of_img_columns(dataframe)
+            types[Values.TXT] = txt_columns > 0
+            types[Values.IMG] = img_columns > 0
+            types[Values.CAT] = (
+                dataframe.select_dtypes("O").shape[1]
+                - (txt_columns + img_columns)
+                + dataframe.select_dtypes("boolean").shape[1]
```

Comment:
could you clean this up somehow? it looks a little messy as is. 

maybe create variables for num_obj_columns and num_boolean_columns.